std/algebra/emulated/fields_bw6761/e6.go

```diff
@@ -85,13 +86,18 @@ func (e Ext6) Mul(x, y *E6) *E6 {
 }
 
 func (e Ext6) Square(x *E6) *E6 {
-
-	x = e.Reduce(x)
 	//Algorithm 22 from https://eprint.iacr.org/2010/354.pdf
+	x = e.Reduce(x)
 	c0 := e.Ext3.Sub(&x.B0, &x.B1)
-	c3 := e.Ext3.MulByNonResidue(&x.B1)
-	c3 = e.Ext3.Neg(c3)
-	c3 = e.Ext3.Add(&x.B0, c3)
+	c3 := &E3{
+		A0: x.B1.A2,
+		A1: x.B1.A0,
+		A2: x.B1.A1,
+	}
+	c3.A0 = *e.fp.MulConst(&c3.A0, big.NewInt(4))
```

Comment:
Here it is ok to do `c3.A0 = *e.fp.MulConst(&c3.A0, big.NewInt(4))` (overwrite `c3.A0`) as `c3.A0` is not part of a multiplication check right now, but long-term may be error prone. It would be better to do
```go
d0 := e.fp.MulConst(&x.B1.A2, big.NewInt(4))
d0 = e.fp.Add(&X.B0.A0, d0)
d1 := e.fp.Sub(&x.B0.A1, &x.B1.A0)
d2 := e.fp.Sub(&x.B0.A2, &x.B1.A1)
c3 = &E3{
    A0: *d0,
    A1: *d1,
    A2: *d2,
} 
```
instead to avoid bugs in the future.

Reply:
Got it! anyway this will be superseded by https://github.com/Consensys/gnark/pull/1126